### PR TITLE
Enhance plato implementation

### PR DESCRIPTION
**Pull request by valeriansaliou**
Improvement of #43

### CODE REVIEW

**Comment by valeriansaliou** [comment]
All done, this implementation relies on the plato library API in a more Node-ish way. No more command needed.

It's also stricter on checking, as I mentioned yesterday on other related threads.

**Comment by valeriansaliou** [comment]
Also added a tiny console message on david success. Much better this way than to leave a blank output in case of success, gets the CI console output to more clarity.

File: src/interface/david.js [hunk 1]
```diff
@@ -55,6 +55,8 @@ module.exports = function (debug) {
         return f(new Error('At least one dependency needs to be updated'));
       }
 
+      console.log('All dependencies are up-to-date');
```
Comment (FGRibreau) [inline src/interface/david.js]: :+1:

**Comment by FGRibreau** [comment]
@valeriansaliou I got a:

```
[plato]
All files above maintainability threshold [NaN%]
```

```
src/interface/plato.js
  line 20  col 9   The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.
  line 37  col 69  Missing semicolon.
  line 39  col 22  'i' is already defined.
  line 39  col 11  The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.
  line 4   col 5   'p' is defined but never used.
```

I fixed this, refactored plato.js ([see the changes here](https://github.com/FGRibreau/check-build/commit/5107054250859cc2512b7db018bc03646b450978)) updated readme and then release a new version once its read. (see https://github.com/FGRibreau/check-build/pull/45 )

**Comment by valeriansaliou** [comment]
Oh crap, didn't check for those cases. I thought we could always rely on a properly formatted configuration, since the developer using the lib should know what he's doing.

Anyway, thanks for the merge. Cheers!

**Comment by FGRibreau** [comment]
No problem, thanks again for your Plato integration, have a nice day Valérian!

**Pull request merged**